modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -357,4 +368,46 @@ Map<String, Long> toMap() {
             return results;
         }
     }
+
+    private static final TimeValue RETRY_STATS_WINDOW = TimeValue.timeValueMinutes(5);
+
+    private static class AwsApiRetryStats {
+        private final LongAdder requests = new LongAdder();
+        private final LongAdder retries = new LongAdder();
+        private final LongAdder throttles = new LongAdder();
+    }
+
+    private volatile AwsApiRetryStats awsApiRetryStats = new AwsApiRetryStats();
+
+    private void maybeLogRetryStats() {
+        var stats = awsApiRetryStats;
+        awsApiRetryStats = new AwsApiRetryStats();
```

Comment:
Nit: maybe we can just reset the LongAdders instead of recreating them.

Reply:
I have not found a good way to reset all counters at once without synchronizing